packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -7,14 +7,17 @@
 
 import * as fs from 'graceful-fs';
 import pnpResolver from 'jest-pnp-resolver';
-import {sync as resolveSync} from 'resolve';
+import {AsyncOpts, SyncOpts, sync as resolveSync} from 'resolve';
 import type {Config} from '@jest/types';
 import {tryRealpath} from 'jest-util';
+// QUESTION: What is a better way to import this?
```

Comment:
Hmmmm, unfortunately your suggestion doesn't seem to work, and instead I get:

```
TypeError: (0 , _resolve.default) is not a function
```

Reply:
weird, I'm not getting that... Can you commit and push it?

Reply:
I got closer.  Using:

```
const resolveAsync = require('resolve/lib/async');
```

works correctly, but this does not:

```
const resolveAsync = require('resolve');
```

Feel free to try it out in the branch, remove the `/lib/async` and you'll get an error about `from` not being defined.  It's not a great error, but it's because `resolveAsync` isn't an actual function when imported that way, it's an object, and so the callback doesn't get called.

Reply:
pushed a commit with those imports. The issue was the mock in the test